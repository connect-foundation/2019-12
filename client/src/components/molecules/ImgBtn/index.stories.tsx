import React from 'react';
import { text } from '@storybook/addon-knobs';

import ImgBtn from '.';

export default {
  title: 'Molecules / ImgBtn',
};

export const imgBtn: React.FC = () => (
  <ImgBtn
    src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
    alt={text('alt', 'img btn alt')}
    href="http://naver.com"
  />
);
