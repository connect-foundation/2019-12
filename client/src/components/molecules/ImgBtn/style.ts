import styled from 'styled-components';
import { palette } from 'styled-tools';

import Btn from 'components/atoms/Btn';
import Img from 'components/atoms/Img';

export const Button = styled(Btn)`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 0;
  background: none;
  margin: auto;
  padding: 0;
`;

export const Image = styled(Img)`
  &:hover {
    opacity: ${palette('opacityscale', 1)};
    transition: opacity 1s;
  }
`;
