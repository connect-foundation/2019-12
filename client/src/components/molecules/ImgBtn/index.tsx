import React from 'react';

import * as S from './style';

interface Props {
  /** img src */
  src: string;
  /** img alt */
  alt: string;
  /** react-router/Link 사용 (내부적인 routing)*/
  to?: string;
  /** 외부 링크 */
  href?: string;
}

function ImgBtn({ src, alt, ...props }: Props): React.ReactElement {
  return (
    <S.Btn {...props}>
      <S.Img alt={alt} src={src} />
    </S.Btn>
  );
}

export default ImgBtn;
