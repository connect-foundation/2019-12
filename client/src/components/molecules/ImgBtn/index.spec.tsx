import React from 'react';
import { mount } from 'enzyme';
import ImgBtn from '.';

describe('Molecules / ImgBtn', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(
      <ImgBtn
        src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
        alt={'img btn alt'}
        href="http://naver.com"
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
