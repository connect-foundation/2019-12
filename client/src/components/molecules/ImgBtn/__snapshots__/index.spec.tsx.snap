// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Img [SNAPSHOT] 렌더링 1`] = `
.c1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 18rem;
  height: 4rem;
  line-height: 1.4rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  pointer-events: auto;
  -webkit-user-drag: none;
  display: inline-block;
  padding: 1.2rem 2rem;
  border-radius: 0.3rem;
  border-color: transparent;
  outline: none;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: opacity ease 0.2s,background-color ease 0.2s,color ease 0.2s;
  transition: opacity ease 0.2s,background-color ease 0.2s,color ease 0.2s;
  border-style: solid;
  border-width: thin;
  line-height: 0;
}

.c3 {
  width: inherit;
  height: inherit;
  object-fit: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  background: none;
  margin: auto;
  padding: 0;
}

.c2:hover {
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

<ImgBtn
  alt="main banner img"
  src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
>
  <Styled(Btn)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  border: 0;
  background: none;
  margin: auto;
  padding: 0;
",
            ],
          },
          "displayName": "Styled(Btn)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <Btn
        className="c0"
      >
        <styled.button
          className="c0"
          customprops={
            Object {
              "fit": false,
              "grow": false,
            }
          }
          styletype="primary"
        >
          <StyledComponent
            className="c0"
            customprops={
              Object {
                "fit": false,
                "grow": false,
              }
            }
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    "
  flex-grow: ",
                    [Function],
                    ";
  width: ",
                    [Function],
                    ";
  height: ",
                    [Function],
                    ";
  line-height: 1.4rem;
  background-color: ",
                    [Function],
                    ";
  flex-shrink: 0;
  cursor: ",
                    [Function],
                    ";
  pointer-events: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  -webkit-user-drag: none;
  display: inline-block;
  padding: 1.2rem 2rem;
  border-radius: 0.3rem;
  border-color: ",
                    [Function],
                    ";
  outline: none;
  text-align: center;
  user-select: none;
  transition: opacity ease 0.2s, background-color ease 0.2s, color ease 0.2s;
  border-style: solid;
  border-width: thin;

  &:hover {
    opacity: ",
                    [Function],
                    ";
    ",
                    [Function],
                    "
  }
",
                    "
  line-height: 0;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            styletype="primary"
          >
            <button
              className="c1 c0"
            >
              <Styled(Img)
                alt="main banner img"
                src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
              >
                <StyledComponent
                  alt="main banner img"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  &:hover {
    opacity: ",
                          [Function],
                          ";
    transition: opacity 1s;
  }
",
                        ],
                      },
                      "displayName": "Styled(Img)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                >
                  <Img
                    alt="main banner img"
                    className="c2"
                    src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                  >
                    <styled.img
                      alt="main banner img"
                      src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                    >
                      <StyledComponent
                        alt="main banner img"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bxivhb",
                              "isStatic": false,
                              "lastClassName": "c3",
                              "rules": Array [
                                "
  width: inherit;
  height: inherit;
  object-fit: inherit;
",
                              ],
                            },
                            "displayName": "styled.img",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bxivhb",
                            "target": "img",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                      >
                        <img
                          alt="main banner img"
                          className="c3"
                          src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                        />
                      </StyledComponent>
                    </styled.img>
                  </Img>
                </StyledComponent>
              </Styled(Img)>
            </button>
          </StyledComponent>
        </styled.button>
      </Btn>
    </StyledComponent>
  </Styled(Btn)>
</ImgBtn>
`;

exports[`Molecules / ImgBtn [SNAPSHOT] 렌더링 1`] = `
.c1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 18rem;
  height: 4rem;
  line-height: 1.4rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  cursor: pointer;
  pointer-events: auto;
  -webkit-user-drag: none;
  display: inline-block;
  padding: 1.2rem 2rem;
  border-radius: 0.3rem;
  border-color: transparent;
  outline: none;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: opacity ease 0.2s,background-color ease 0.2s,color ease 0.2s;
  transition: opacity ease 0.2s,background-color ease 0.2s,color ease 0.2s;
  border-style: solid;
  border-width: thin;
}

.c3 {
  width: inherit;
  height: inherit;
  object-fit: inherit;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  background: none;
  margin: auto;
  padding: 0;
}

.c2:hover {
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
}

<ImgBtn
  alt="img btn alt"
  href="http://naver.com"
  src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
>
  <Styled(Btn)
    href="http://naver.com"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  align-items: center;
  border: 0;
  background: none;
  margin: auto;
  padding: 0;
",
            ],
          },
          "displayName": "Styled(Btn)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      href="http://naver.com"
    >
      <Btn
        className="c0"
        href="http://naver.com"
      >
        <styled.a
          className="c0"
          customprops={
            Object {
              "fit": false,
              "grow": false,
            }
          }
          href="http://naver.com"
          styletype="primary"
        >
          <StyledComponent
            className="c0"
            customprops={
              Object {
                "fit": false,
                "grow": false,
              }
            }
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    "
  flex-grow: ",
                    [Function],
                    ";
  width: ",
                    [Function],
                    ";
  height: ",
                    [Function],
                    ";
  line-height: 1.4rem;
  background-color: ",
                    [Function],
                    ";
  flex-shrink: 0;
  cursor: ",
                    [Function],
                    ";
  pointer-events: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  -webkit-user-drag: none;
  display: inline-block;
  padding: 1.2rem 2rem;
  border-radius: 0.3rem;
  border-color: ",
                    [Function],
                    ";
  outline: none;
  text-align: center;
  user-select: none;
  transition: opacity ease 0.2s, background-color ease 0.2s, color ease 0.2s;
  border-style: solid;
  border-width: thin;

  &:hover {
    opacity: ",
                    [Function],
                    ";
    ",
                    [Function],
                    "
  }
",
                    "
",
                  ],
                },
                "displayName": "styled.a",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href="http://naver.com"
            styletype="primary"
          >
            <a
              className="c1 c0"
              href="http://naver.com"
            >
              <Styled(Img)
                alt="img btn alt"
                src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
              >
                <StyledComponent
                  alt="img btn alt"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
  &:hover {
    opacity: ",
                          [Function],
                          ";
    transition: opacity 1s;
  }
",
                        ],
                      },
                      "displayName": "Styled(Img)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                >
                  <Img
                    alt="img btn alt"
                    className="c2"
                    src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                  >
                    <styled.img
                      alt="img btn alt"
                      src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                    >
                      <StyledComponent
                        alt="img btn alt"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bxivhb",
                              "isStatic": false,
                              "lastClassName": "c3",
                              "rules": Array [
                                "
  width: inherit;
  height: inherit;
  object-fit: inherit;
",
                              ],
                            },
                            "displayName": "styled.img",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-bxivhb",
                            "target": "img",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                      >
                        <img
                          alt="img btn alt"
                          className="c3"
                          src="https://sprint.kr.object.ncloudstorage.com/tempimages/main-banner-1"
                        />
                      </StyledComponent>
                    </styled.img>
                  </Img>
                </StyledComponent>
              </Styled(Img)>
            </a>
          </StyledComponent>
        </styled.a>
      </Btn>
    </StyledComponent>
  </Styled(Btn)>
</ImgBtn>
`;
